lib/blacklight/search_state.rb

```diff
@@ -46,9 +46,7 @@ def reset(params = nil)
     # to provide more interesting routing to
     # documents
     def url_for_document(doc, options = {})
-      if respond_to?(:blacklight_config) &&
-          blacklight_config.show.route &&
-          (!doc.respond_to?(:to_model) || doc.to_model.is_a?(SolrDocument))
+      if blacklight_config.show.route || (doc.respond_to?(:to_model) && doc == doc.to_model)
```

Comment:
I think this `||` must still be an `&&`. Because the tests fail due to `blacklight_config.show.route` being nil